packages/@aws-cdk/aws-cloudfront/test/test.basic.ts

```diff
@@ -325,4 +326,62 @@ export = {
     test.done();
   },
 
+  'throws if certificate is not in us-east-1 - fromCertificateArn'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const sourceBucket = new s3.Bucket(stack, 'Bucket');
+    const certificate = certificatemanager.Certificate.fromCertificateArn(
+        stack, 'EuCertificate', 'arn:aws:acm:eu-west-1:1234567890:certificate/testACM'
+    );
+
+    // WHEN
+    const toThrow = () => {
+      new CloudFrontWebDistribution(stack, 'AnAmazingWebsiteProbably', {
+          originConfigs: [
+          {
+            s3OriginSource: {s3BucketSource: sourceBucket},
+            behaviors: [{isDefaultBehavior: true}]
+          }
+        ],
+        aliasConfiguration: {
+          names: ['www.example.com'],
+          acmCertRef: certificate.certificateArn
+        }
+      });
+    };
+
+    // THEN
+    test.throws(() => toThrow(), /acmCertificateArn must be in the 'us-east-1' region, got 'eu-west-1'/);
+    test.done();
+  },
+
+  'throws if certificate arn is invalid - constructor'(test: Test) {
+    // GIVEN
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'RegionStack', {env: {region: 'eu-west-3'}});
+    const sourceBucket = new s3.Bucket(stack, 'Bucket');
+    const certificate = new certificatemanager.Certificate(stack, 'TestCertificate', {
```

Comment:
The `region` token is unresolved when I `Arn.parse` it. Is there a way to retrieve it through the stack?

Reply:
This is because `certificate.certificateArn` returns a token that resolves to an `Fn::GetAtt`, so `parseArn` is unable to actually parse the ARN during synthesis but rather returns some cloudformation intrinsic function expressions that will parse the ARN during deployment.

Specifically for the test, you can just use `Certificate.from` to import from a concrete ARN and then `parseArn` will return the actual value. 

But if I understand correctly (and I am not 100% sure), is it only possible to define certificates in stacks that are deployed to us-east-1? If this is indeed the case, we should add a validation in the constructor of `Certificate` which goes to the stack and verifies that it is in us-east-1:

```ts
const region = Stack.of(this).region;
if (!Token.isUnresolved(region) && region !== 'us-east-1') { throw Error('boom'); }
```

Reply:
Thanks, I'll look into it this week.

Your last paragraph, if I understand it correctly, isn't true though. ACM is available in all regions, but CloudFront is only in `us-east-1`, and you can't use certificates across regions.

So for example, if you want to deploy an ELB in `eu-west-1` with an HTTPS listener, and a CloudFront distribution, you'll need to generate two certificates, one for the ELB region, and one in `us-east-1`.

Reply:
Got it. Yeah, I figured I am missing something :-). Thanks for clarifying.

Reply:
I must have misunderstood something in your previous comment about parsing the region.

I tried the following for the failing `new certificatemanager.Certificate` test:
```ts
// ${Token[TOKEN.74]}
let {region} = cdk.Arn.parse(acmCertificateArn);

if (cdk.Token.isUnresolved(region)) {
  const certificate = certificatemanager.Certificate.fromCertificateArn(scope, 'cert', acmCertificateArn);
  // ${Token[TOKEN.74]}
  region = cdk.Arn.parse(certificate.certificateArn).region;
}
```

The region is still an unresolved token. Am I missing something?

Reply:
I understand the limitation better now, thanks 👍